src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryItem.cs

```diff
@@ -0,0 +1,383 @@
+        public override int GetHashCode()
+        {
+            return Basemap?.Name?.GetHashCode() ?? -1;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this basemap is a valid selection.
+        /// </summary>
+        public bool IsValid
+        {
+            get => _isValid;
+            private set
+            {
+                if (_isValid != value)
+                {
+                    _isValid = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsValid)));
+                }
+            }
+        }
+
+        #if XAMARIN_FORMS
+        private bool _isSelected;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to show item as selected. Selection property used to assist manual list selection implementation for Xamarin.Forms.
+        /// </summary>
+        /// <remarks>
+        /// Native listview selection can't be used because of many outstanding Xamarin.Forms,
+        /// specifically affecting UWP: e.g. https://github.com/xamarin/Xamarin.Forms/issues/11405, https://github.com/xamarin/Xamarin.Forms/issues/12491.
+        /// </remarks>
+        public bool IsSelected
+        {
+            get => _isSelected;
+            set
+            {
+                if (value != _isSelected)
+                {
+                    _isSelected = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsSelected)));
+                }
+            }
+        }
+        #endif
+
+        /// <summary>
+        /// Gets the basemap associated with this basemap item.
+        /// </summary>
+        public Basemap Basemap { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the thumbnail to display for this basemap item.
+        /// </summary>
+        public RuntimeImage? Thumbnail
+        {
+            get
+            {
+                if (_thumbnailOverride != null)
+                {
+                    return _thumbnailOverride;
+                }
+
+                return Basemap.Item?.Thumbnail;
+            }
+
+            set
+            {
+                if (_thumbnailOverride != value)
+                {
+                    _thumbnailOverride = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Thumbnail)));
+                    _ = LoadImage();
+                }
+            }
+        }
+
+#if XAMARIN_FORMS || (!__IOS__ && !__ANDROID__)
+        /// <summary>
+        /// Gets the thumbnail in a format that is easily displayable in a view.
+        /// </summary>
+        public ImageSource? ThumbnailImageSource
+        {
+            get => _thumbnailImageSource;
+            private set
+            {
+                _thumbnailImageSource = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(ThumbnailImageSource)));
+            }
+        }
+#endif
```

Comment:
If moved to a common class, we might need to use a converter here instead to bind from Thumbnail